src/Ads/AssetSuggestionsService.php

```diff
@@ -152,23 +200,33 @@ protected function get_term_assets( int $id ): array {
 			$attachments_ids = [ ...$this->get_post_image_attachments( [ 'post_parent__in' => $posts_ids_assigned_to_term ] ), ...$attachments_ids ];
 		}
 
+		$attachments_ids  = array_slice( $attachments_ids, 0, self::DEFAULT_MAXIMUM_MARKETING_IMAGES );
 		$marketing_images = $this->get_url_attachments_by_ids( $attachments_ids );
```

Comment:
💅  

Related to https://github.com/woocommerce/google-listings-and-ads/pull/1804/files#r1045420662

We can avoid verbosity implementing limiting the ids size in  `get_url_attachments_by_ids` scope